homeassistant/components/toon/config_flow.py

```diff
@@ -1,166 +1,92 @@
-        )
+        This step is merely here to trigger "discovery" when the `toon`
+        integration is listed in the user configuration, or when migrating from
+        the version 1 schema.
+        """
+        await self._async_handle_discovery_without_unique_id()
+        return await self.async_step_user()
 
-    async def async_step_display(self, user_input=None):
-        """Select Toon display to add."""
+    async def async_step_agreement(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Select Toon agreement to add."""
+        if len(self.agreements) == 1:
+            return await self._create_entry(self.agreements[0])
 
-        if not self.displays:
-            return self.async_abort(reason="no_displays")
+        agreements_list = [
+            f"{agreement.agreement_id} - {agreement.street} {agreement.house_number}, {agreement.city}"
+            for agreement in self.agreements
+        ]
 
         if user_input is None:
-            return await self._show_display_form()
-
-        if user_input[CONF_DISPLAY] in configured_displays(self.hass):
-            return await self._show_display_form({"base": "display_exists"})
-
-        app = self.hass.data.get(DATA_TOON_CONFIG, {})
-        try:
-            await self.hass.async_add_executor_job(
-                partial(
-                    Toon,
-                    self.username,
-                    self.password,
-                    app[CONF_CLIENT_ID],
-                    app[CONF_CLIENT_SECRET],
-                    tenant_id=self.tenant,
-                    display_common_name=user_input[CONF_DISPLAY],
-                )
+            return self.async_show_form(
+                step_id="agreement",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_AGREEMENT): vol.In(agreements_list)}
+                ),
             )
 
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Unexpected error while authenticating")
-            return self.async_abort(reason="unknown_auth_fail")
+        _LOGGER.error(user_input)
+        agreement_index = agreements_list.index(user_input[CONF_AGREEMENT])
+        return await self._create_entry(self.agreements[agreement_index])
+
+    async def _create_entry(self, agreement: Agreement) -> Dict[str, Any]:
+        await self.async_set_unique_id(agreement.agreement_id)
+        self._abort_if_unique_id_configured()
 
+        self.data[CONF_AGREEMENT_ID] = agreement.agreement_id
         return self.async_create_entry(
-            title=user_input[CONF_DISPLAY],
-            data={
-                CONF_USERNAME: self.username,
-                CONF_PASSWORD: self.password,
-                CONF_TENANT: self.tenant,
-                CONF_DISPLAY: user_input[CONF_DISPLAY],
-            },
+            title=f"{agreement.agreement_id} - {agreement.street} {agreement.house_number}, {agreement.city}",
```

Comment:
Is agreement ID so important that it needs to be in the title?

Reply:
as in, would it mean anything to a normal user?

Reply:
I guess not, lets remove it indeed 👍

Reply:
Can we also remove it from the agreement picker that happens after authentication ?